File: 2024/Day_01/part01.cpp
#include <iostream>
#include <vector>
#include <fstream>
#include <algorithm>
using namespace std;

void readData(vector<int> &leftCol, vector<int> &rightCol) {
    ifstream file("test.txt");
    if(!file.is_open()) {
        cerr << "Error Opening File" << endl;
        exit(-1);
    }

    int left, right;
    while (file >> left >> right) {
        leftCol.push_back(left);
        rightCol.push_back(right);
    }

    file.close();
}

int main() {
    vector<int> leftCol;
    vector<int> rightCol;
    readData(leftCol, rightCol);

    sort(leftCol.begin(), leftCol.end());
    sort(rightCol.begin(), rightCol.end());

    int totalDist = 0;
    for (int i = 0; i < leftCol.size(); i++) {
        totalDist += abs(leftCol[i]-rightCol[i]);
    }
    cout << totalDist << endl;

    return 0;
}

File: 2024/Day_01/part01.py
import numpy as np

# read data from file
data = np.loadtxt('test.txt').astype(int)
left_col = data[:, 0]  
right_col = data[:, 1]

# calculates total distance
left_col = np.sort(left_col)
right_col = np.sort(right_col)
total_dist = abs(left_col - right_col)

print(np.sum(total_dist))